v2/api/containerservice/v1api20230201/storage/compat/cluster_upgrade_settings_types.go

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package compat
+
+import (
+	v20231001s "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20231001/storage"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+var _ augmentConversionForUpgradeOverrideSettings = &UpgradeOverrideSettings{}
+
+const ignoreKubernetesDeprecations = "IgnoreKubernetesDeprecations"
+
+func (settings *UpgradeOverrideSettings) AssignPropertiesFrom(src *v20231001s.UpgradeOverrideSettings) error {
+	// If the GA version has ForceUpgrade true, the preview version needs to add "IgnoreKubernetesDeprecations"
+	if src.ForceUpgrade != nil && *src.ForceUpgrade {
+		settings.ControlPlaneOverrides = append(settings.ControlPlaneOverrides, ignoreKubernetesDeprecations)
+	}
+
+	return nil
+}
+
+func (settings *UpgradeOverrideSettings) AssignPropertiesTo(dest *v20231001s.UpgradeOverrideSettings) error {
+	// If the preview version has "IgnoreKubernetesDeprecations" in ControlPlaneOverrides, the GA version needs to have
+	// ForceUpgrade true, otherwise false.
+	dest.ForceUpgrade = to.Ptr(false)
```

Comment:
Does this work for round-trip cases? Do we need to deal w/ `nil`? So something like:
- If nil, then `ForceUpgrade` nil.
- If empty, false
- If populated w/ IgnoreKubernetesDeprecations, true

Not sure it's needed just checking.

Reply:
And I wrote unit tests to check all the nil cases.